apis/kubekey/v1alpha1/network_types.go

```diff
@@ -32,7 +32,12 @@ type CalicoCfg struct {
 }
 
 type FlannelCfg struct {
-	BackendMode string `yaml:"backendMode" json:"backendMode,omitempty"`
+	BackendMode string     `yaml:"backendMode" json:"backendMode,omitempty"`
+	Backend     BackendCfg `yaml:"backend" json:"backend,omitempty"`
+}
+type BackendCfg struct {
```

Comment:
According to this struct, we need to write our config yaml file like this:
```
network:
  plugin: flannel
  kubePodsCIDR: 10.233.64.0/18
  kubeServiceCIDR: 10.233.0.0/18
  flannel:
    backendMode: vxlan
    backend:
      directRouting: true
```
So, I think it might be better just like the @shake issue comment #606:
```
network:
  plugin: flannel
  flannel:
    backendMode: vxlan
    directrouting: true
```

Reply:
I tested it successfully like this.

```
network:
  plugin: flannel
  kubePodsCIDR: 10.233.64.0/18
  kubeServiceCIDR: 10.233.0.0/18
  flannel:
    backend:
      type: vxlan
      directRouting: true
```

Reply:
Your code is definitely correct.

Reply:
The result of kk running is definitely correct. 

But according to your config yaml file will assign `vxlan` to the `type` field and the `BackendMode` is an empty string. 

In the `pkg/plugins/network/flannel/tmpl.go` the variable `{{ .BackendMode }}` value is `vxlan` caused by there is a default const named by `DefaultBackendMode` assigned to `BackendMode` when kk init.

In terms of convenience, i think it will more simple to confiure by the second way.